src/components/Login/LoginContainer.tsx

```diff
@@ -191,11 +201,15 @@ export const InitialisationContainer: FunctionComponent<NavigationProps> = ({
       try {
         const { key, endpoint } = decodeQr(qrCode);
         Vibration.vibrate(50);
+        if (!RegExp(DOMAIN_CHECK).test(endpoint))
+          throw new Error("Invalid endpoint");
         setCodeKey(key);
         setEndpointTemp(endpoint);
         setIsLoading(false);
         setLoginStage("MOBILE_NUMBER");
       } catch (e) {
+        const error = new Error(`onBarCodeScanned ${e}`);
+        Sentry.captureException(error);
```

Comment:
just to clarify! for this portion, after throwing error `Invalid endpoint`, is it intended for the user to just see `Invalid QR Code` as the error message?

Reply:
ahh I see what you mean! Previously the `Invalid QR Code` was there before this haha xD Maybe we can choose Invalid QR code as the only error thrown?